src/Pos/Block/Network/Announce.hs

```diff
@@ -87,13 +88,21 @@ handleHeadersCommunication conv _ = do
         logDebug $ sformat ("Got request on handleGetHeaders: "%build) mgh
         ifM recoveryInProgress onRecovery $ do
             headers <- case (mghFrom,mghTo) of
-                ([], Nothing) -> Right . one <$> DB.getTipBlockHeader @ssc
+                ([], Nothing) -> Right . one <$> getLastMainBlock
                 ([], Just h)  ->
                     maybeToRight "getBlockHeader returned Nothing" . fmap one <$>
                     DB.getBlockHeader @ssc h
                 (c1:cxs, _)   -> getHeadersFromManyTo (c1:|cxs) mghTo
             either onNoHeaders handleSuccess headers
   where
+    -- retrieves last main block if there's any
+    getLastMainBlock :: m (BlockHeader ssc)
+    getLastMainBlock = do
+        (tip :: Block ssc) <- DB.getTipBlock @ssc
+        let tipHash = tip ^. blockHeader
+        case tip of
+            Left _  -> fromMaybe tipHash <$> DB.getBlockHeader (tip ^. prevBlockL)
```

Comment:
It's not the first genesis block in a normal environment, first. Second, previous implementation would do that anyway.